src/agent/src/image_rpc.rs

```diff
@@ -0,0 +1,160 @@
+
+        // Define the target transport and path for the OCI image, without signature
+        let oci_path = format!("/tmp/{}/image_oci:latest", cid);
+        let target_path_oci = format!("oci://{}", oci_path);
+
+        fs::create_dir_all(&manifest_path)?;
+        fs::create_dir_all(&oci_path)?;
+
+        info!(sl!(), "Attempting to pull image {}...", &source_image);
+
+        let mut pull_command = Command::new(SKOPEO_PATH);
+        pull_command
+            .arg("--insecure-policy")
+            .arg("copy")
+            .arg(source_image)
+            .arg(&target_path_manifest);
+
+        if let Some(source_creds) = source_creds {
+            pull_command.arg("--src-creds").arg(source_creds);
+        }
+
+        let status: ExitStatus = pull_command.status()?;
+
+        if !status.success() {
+            return Err(anyhow!(format!("failed to pull image: {:?}", status)));
+        }
+
+        // Copy image from one local file-system to another
+        // Resulting image is still stored in manifest format, but no longer includes the signature
+        // The image with a signature can then be unpacked into a bundle
+        let status: ExitStatus = Command::new(SKOPEO_PATH)
+            .arg("--insecure-policy")
+            .arg("copy")
+            .arg(&target_path_manifest)
+            .arg(&target_path_oci)
+            .arg("--remove-signatures")
+            .status()?;
+
+        if !status.success() {
+            return Err(anyhow!(format!("failed to copy image: {:?}", status)));
+        }
+
+        // To save space delete the manifest.
+        // TODO LATER - when verify image is added, this will need moving the end of that, if required
+        fs::remove_dir_all(&manifest_path)?;
+        Ok(())
+    }
+
+    fn unpack_image(cid: &str) -> Result<()> {
+        let source_path_oci = format!("/tmp/{}/image_oci:latest", cid);
+        let target_path_bundle = format!("{}{}{}", CONTAINER_BASE, "/", cid);
+
+        info!(sl!(), "cid is {:?}", cid);
+        info!(sl!(), "target_path_bundle is {:?}", target_path_bundle);
+
+        // Unpack image
+
+        let status: ExitStatus = Command::new(UMOCI_PATH)
+            .arg("--verbose")
+            .arg("unpack")
+            .arg("--image")
+            .arg(&source_path_oci)
+            .arg(&target_path_bundle)
+            .status()?;
+
+        if !status.success() {
+            return Err(anyhow!(format!("failed to unpack image: {:?}", status)));
+        }
+
+        // To save space delete the oci image after unpack
+        fs::remove_dir_all(&source_path_oci)?;
+
+        Ok(())
+    }
+
+    async fn pull_image(&self, req: &image::PullImageRequest) -> Result<String> {
+        let image = req.get_image();
+        let mut cid = req.get_container_id();
+
+        if cid.len() == 0 {
+            let v: Vec<&str> = image.rsplit('/').collect();
+            if v[0].len() != 0 {
+                cid = v[0]
+            }
```

Comment:
It looks like `cid` could still be blank, so I think we need a check after this block so we can return an error in that case.

Reply:
I was a bit lazy here and wanted to rely on the skopeo return error later.